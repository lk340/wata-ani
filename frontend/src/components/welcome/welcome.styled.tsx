import styled, { css, createGlobalStyle } from "styled-components";
import { animated } from "react-spring";

export const Hello = styled("div").attrs(() => ({
	"data-testid": "hello",
}))``;
