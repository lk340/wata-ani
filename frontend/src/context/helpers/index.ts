import { createUseContext } from "./create-use-context";

export { createUseContext };
