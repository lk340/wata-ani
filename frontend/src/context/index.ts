// ================= //
// ↓↓↓ Utilities ↓↓↓ //
// ================= //

import * as UserAgent from "@/context/utilities/user-agent";
import * as Location from "@/context/utilities/location";
import * as Windows from "@/context/utilities/windows";
import * as Scroll from "@/context/utilities/scroll";
import * as Theme from "@/context/utilities/theme";
import * as Mouse from "@/context/utilities/mouse";

// ============ //
// ↓↓↓ Data ↓↓↓ //
// ============ //

import * as AuthForm from "@/context/data/auth-form";
import * as Navbar from "@/context/data/navbar";
import * as Pagination from "@/context/data/pagination";
import * as NavbarOptionsCreate from "@/context/data/navbar-options-create";
import * as NavbarOptionsSettings from "@/context/data/navbar-options-settings";

// =============== //
// ↓↓↓ Exports ↓↓↓ //
// =============== //

export {
	UserAgent,
	Location,
	Windows,
	Scroll,
	Theme,
	Mouse,
	AuthForm,
	Navbar,
	Pagination,
	NavbarOptionsCreate,
	NavbarOptionsSettings,
};
