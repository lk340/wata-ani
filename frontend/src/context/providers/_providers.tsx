import * as React from "react";

import * as Context from "@/context";

export const Providers: React.FC<{ children: React.ReactNode }> = ({ children }) => {
	return (
		<>
			<Context.Theme.Provider>
				<Context.Window.Provider>
					<Context.Scroll.Provider>
						<Context.Mouse.Provider>
							<Context.Location.Provider>
								<Context.Auth.Provider>
									<Context.UserAgent.Provider>
										<Context.Welcome.Provider>{children}</Context.Welcome.Provider>
									</Context.UserAgent.Provider>
								</Context.Auth.Provider>
							</Context.Location.Provider>
						</Context.Mouse.Provider>
					</Context.Scroll.Provider>
				</Context.Window.Provider>
			</Context.Theme.Provider>
		</>
	);
};
