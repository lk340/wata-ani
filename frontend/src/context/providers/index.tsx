import * as React from "react";

import * as Context from "@/context";

export const Providers: React.FC<{ children: React.ReactNode }> = ({ children }) => {
	return (
		<>
			<Context.Theme.Provider>
				<Context.Windows.Provider>
					<Context.Scroll.Provider>
						<Context.Mouse.Provider>
							<Context.Location.Provider>
								<Context.UserAgent.Provider>
									<Context.AuthForm.Provider>
										<Context.Navbar.Provider>
											<Context.Pagination.Provider>
												<Context.NavbarOptionsCreate.Provider>
													<Context.NavbarOptionsSettings.Provider>
														{children}
													</Context.NavbarOptionsSettings.Provider>
												</Context.NavbarOptionsCreate.Provider>
											</Context.Pagination.Provider>
										</Context.Navbar.Provider>
									</Context.AuthForm.Provider>
								</Context.UserAgent.Provider>
							</Context.Location.Provider>
						</Context.Mouse.Provider>
					</Context.Scroll.Provider>
				</Context.Windows.Provider>
			</Context.Theme.Provider>
		</>
	);
};

// const providers = [
// 	Context.Theme.Provider,
// 	Context.Window.Provider,
// 	Context.Scroll.Provider,
// 	Context.Mouse.Provider,
// 	Context.Location.Provider,
// 	Context.Auth.Provider,
// 	Context.UserAgent.Provider,
// 	Context.Filter.Provider,
// 	Context.Sort.Provider,
// 	Context.Demo.Provider,
// 	Context.Home.Provider,
// ];

// function generateProvider(providers: React.ReactNode[]) {
// 	if (providers.length === 1) return providers[0];

// 	const lastIndex = providers.length - 1;

// }

// const Children: React.FC<{ children: React.ReactNode }> = ({ children }) => {
// 	return <>{children}</>;
// };
