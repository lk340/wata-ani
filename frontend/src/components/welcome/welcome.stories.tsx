import * as React from "react";

import { Providers } from "@/context/providers";

import { Welcome } from "./welcome";

type Story = {
	component: React.ReactNode;
	title: string;
	excludeStories: RegExp;
};

const story: Story = {
	component: Welcome,
	title: "Welcome",
	// Our exports that end in "Data" are not stories. (i.e. ignore taskData & actionsData)
	excludeStories: /.*Data$/,
};

export default story;

export const taskData = {};

export const Default = () => {
	return (
		<Providers>
			<Welcome />
		</Providers>
	);
};
