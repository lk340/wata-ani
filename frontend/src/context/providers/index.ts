import { Providers } from "./_providers";
export { Providers };
