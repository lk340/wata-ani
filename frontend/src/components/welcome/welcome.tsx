import * as React from "react";

import * as Context from "@/context";

import * as Styled from "./welcome.styled";

export const Welcome = () => {
	const { welcome } = Context.Welcome.useWelcomeContext();

	return <Styled.Hello>{welcome.state.message}</Styled.Hello>;
};
