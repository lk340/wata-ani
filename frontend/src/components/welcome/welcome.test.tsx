import * as React from "react";
import * as Testing from "@testing-library/react";

import { Providers } from "@/context/providers";

import { Welcome } from "./welcome";

function renderWithContext(component: React.ReactNode) {
	return Testing.render(<Providers>{component}</Providers>);
}

describe("Welcome Component", () => {
	it("matches the snapshot", () => {
		const { asFragment } = renderWithContext(<Welcome />);
		expect(asFragment(<Welcome />)).toMatchSnapshot();
	});

	it(`has initial message "Gatsby + TypeScript Starter"`, () => {
		const { getByTestId } = renderWithContext(<Welcome />);
		expect(getByTestId("hello")).toHaveTextContent("Gastby + TypeScript Starter");
	});
});
