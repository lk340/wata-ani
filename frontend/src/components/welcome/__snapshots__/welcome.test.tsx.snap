// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="hello"
  >
    Gastby + TypeScript Starter
  </div>
</DocumentFragment>
`;
